pkg/crc/adminhelper/hosts_windows.go

```diff
@@ -6,7 +6,12 @@ import (
 	"github.com/code-ready/crc/pkg/os/windows/powershell"
 )
 
-func execute(args ...string) error {
+func executePrivileged(args ...string) error {
 	_, _, err := powershell.ExecuteAsAdmin("modifying hosts file", strings.Join(append([]string{goodhostPath}, args...), " "))
 	return err
 }
```

Comment:
Probably better to land first the admin-helper daemon work?

Reply:
admin-helper daemon still require some work: a preflight to install it, define the protocol, etc.

With this PR, we solve it for macOS and Linux. For Windows, the experience is not too bad. There is just one UAC prompt per new route.

This network stack could be GA on macOS/Linux after this PR (+ some minor enhancements, better errors, etc.). 
I think we have the feature parity with the traditional network stack.

Reply:
> This network stack could be GA on macOS/Linux after this PR (+ some minor enhancements, better errors, etc.).
>  I think we have the feature parity with the traditional network stack.

daemon autostart is missing on linux (and on macos setups without the tray running). And if I'm not mistaken this also does not handle routes using non-standard ports? I guess autostart falls under the 'minor enhancements category')? I don't mind the non-standard thing for now, I agree we can push for wider testing after this.

Reply:
`does not handle routes using non-standard ports`

I tried to find examples or code around that but I couldn't find it. The only route I see like this would be the one for Kubernetes API.
Router container only exposes port 80 and 443, not others. I don't really understand how an other port can be used.

Reply:
I thought our own doc had such examples ( https://code-ready.github.io/crc/#deploying-sample-application-with-odo_gsg ) but after being deployed, the app is exposed on port 80/443, not on 8080 as I expected. 
https://docs.okd.io/latest/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-load-balancer.html may be related to this, but is not working with the default networking anyway.